### PR TITLE
nia/docs 0.3.0

### PR DESCRIPTION
**Pull request by lornasong**

### CODE REVIEW

File: website/content/docs/nia/installation/requirements.mdx
Comment (lornasong) [reply website/content/docs/nia/installation/requirements.mdx:12]: Agree, we are actually currently working with the design team to revamp our docs. Welcome your feedback / suggestions and any clarity you are willing to share on how things are confusing so we can improve.

File: website/content/docs/nia/configuration.mdx
Comment (lornasong) [reply website/content/docs/nia/configuration.mdx:41]: You are correct. I'm not sure if you're asking because the language has some ambiguity. Please let me know if there is and happy to fix that.

Looking at consul docs, the file is likely to have a `.hclic` extension. It's no necessary to have that specific extension since we test with `.lic` file extensions. Do you recommend that we mention the expected extension here?

File: website/content/docs/nia/network-drivers/terraform.mdx
Comment (findkim) [reply website/content/docs/nia/network-drivers/terraform.mdx:50]: Resemble. There's additional options that are available in `terraform_provider` and not `provider`.

Terraform config
```hcl
provider {
  // normal Terraform provider options
  address = ""
}
```

CTS config
```hcl
terraform_provider {
  // normal Terraform provider options
  address = ""

  // CTS only
  // load secrets from kv, vault, env to avoid writing values on disk
  task_env {
    "FOO_TOKEN" = "{{ key \"consulkvpath\" }}"
  }
}
```

File: website/content/docs/nia/enterprise/index.mdx
Comment (lornasong) [reply website/content/docs/nia/enterprise/index.mdx:5]: To clarify CTS is offered as an OSS binary and an Enterprise binary, similar to Consul OSS vs. Enterprise. This Enterprise section of the documentation is intended to encapsulate all enterprise related-information, such as licensing and enterprise features. This overview page is intended to give an overview of CTS enterprise - similar to [Consul Enterprise's overview page](https://www.consul.io/docs/enterprise).

Please let me know if that helps clarify the confusion

File: website/content/docs/nia/enterprise/index.mdx
Comment (lornasong) [reply website/content/docs/nia/enterprise/index.mdx:14]: A license is only required for the Enterprise binary of CTS. I think of the /nia/installation/requirements page as a quick start for OSS. What are your thoughts?

File: website/content/docs/nia/configuration.mdx
Comment (lornasong) [reply website/content/docs/nia/configuration.mdx:262]: Yes, there is a CTS enterprise binary. CTS OSS does not require a license

File: website/content/docs/nia/configuration.mdx
Comment (findkim) [reply website/content/docs/nia/configuration.mdx:47]: Thanks for the suggestion @trujillo-adam! Spacing out the note does read better, but there's some limitations around that. The call-out block only supports a contiguous paragraph and even sneaking in `<br>`s break the rendering.

image

Comment (findkim) [reply website/content/docs/nia/configuration.mdx:47]: So to iterate on this, I'm thinking have the first and second paragraphs within the callout, and then the last sentence moved to the `tls` config option. Would this work for you?

```suggest
-> **Note:** Use HTTP/2 to improve Consul-Terraform-Sync performance when communicating with the local Consul process. [TLS/HTTPS](/docs/agent/options) must be configured for the local Consul with the [cert_file](/docs/agent/options#cert_file) and [key_file](/docs/agent/options#key_file) parameters set. For the Consul-Terraform-Sync configuration, set `tls.enabled = true` and set the `address` parameter to the HTTPS URL, e.g., `address = example.consul.com:8501`. If using self-signed certificates for Consul, you will also need to set `tls.verify = false`. 

...

- `tls` - Configure TLS to use a secure client connection with Consul. Using HTTP/2 can solve issues related to hitting Consul's maximum connection limits, as well as improve efficiency when processing many blocking queries. This option is required for Consul-Terraform-Sync when connecting to a [Consul agent with TLS verification enabled for HTTPS connections](/docs/agent/options#verify_incoming).
```

File: website/content/docs/nia/configuration.mdx
Comment (findkim) [reply website/content/docs/nia/configuration.mdx:277]: I agree that we should not include future documentation. We do consider this a special case since it's not common for TFE to release before TFC these days, and are typically released at the same time or TFC first then TFE soon after.

The decision for TFE first was a difficult one for us to work-around and so this callout was added in attempt to address 2 concerns we had:
1. To help with expectations due to the norm of release cadence for TFC and TFE  features
2. The driver is named `terraform_cloud` but only supports TFE would be a non-intuitive user experience.

Does this added context help with why we have these callouts?

File: website/content/docs/nia/configuration.mdx
Comment (lornasong) [reply website/content/docs/nia/configuration.mdx:169]: Thanks for the improvements here. To clarify

> "The default condition is to trigger on changes to task.services"

I'm more specifically trying to say (in a wordy way) that the default condition's default trigger is to trigger on task.services. What do you think about updating the above sentence to "The default values to trigger on are the services configured in task.services"?

File: website/content/docs/nia/configuration.mdx
Comment (lornasong) [reply website/content/docs/nia/configuration.mdx:301]: To clarify, the default behavior (when workspace_prefix is not configured) is for the workspace name to be the same as the task name (no prefix). When workspace_prefix is configured then the workspace name would be the prefix + the task name.

What are your thoughts on updating this to:

>- `workspace_prefix` - (string) Specifies a prefix to prepend to the automatically-generated workspace names used for automation. By default, when `workspace_prefix` is unconfigured, the automatically-generated workspace name is the task name. When `workspace_prefix` is configured, the workspace name will be `<workspace_prefix value><task name>`. For example, if you configure the prefix as "cts_", a task with the name "task_firewall" will have the workspace name of "cts_task_firewall"

File: website/content/docs/nia/network-drivers/terraform-cloud.mdx
Comment (trujillo-adam) [inline website/content/docs/nia/network-drivers/terraform-cloud.mdx:65]:
```suggestion
  -  `services` declares input variables that determine module compatibility with Consul-Terraform Sync (read more on [compatible Terraform modules](/docs/nia/terraform-modules) for more details). 
```
I introduced a typo

File: website/content/docs/nia/configuration.mdx
Comment (findkim) [reply website/content/docs/nia/configuration.mdx:146]: The intention is the files themselves that contains Terraform values.

The abstraction here is residue from us originally writing the docs with flexibility not pinned to Terraform verbiage but that has since changed. We are ok with having Terraform specific verbiage threaded in our docs.

Thanks for pointing out how unclear this portion is for us to revisit! So here's another jab at this section

```suggestion
- `variable_files` - (list[string]) Specifies list of paths to [Terraform variable definition files (`.tfvars`)](https://www.terraform.io/docs/configuration/variables.html#variable-definitions-tfvars-files). The content of these files should consist of only variable name assignments. The variable assignments must match the corresponding variable declarations made available by the Terraform module for the task. 
  - Variables are loaded in the order they appear in the files. Duplicate variables are overwritten with the later value. _Unless specified by the module, configure arguments for Terraform providers using [`terraform_provider` blocks](#terraform-provider)._
    ```hcl
    # example.tfvars
    address_group = "consul-services"
    tags = [
      "consul-terraform-sync",
      "terraform"
    ]
    ```
```

![image](https://user-images.githubusercontent.com/6362111/131385316-34e0096d-bb2f-44d2-b873-8fe7f6c94330.png)

File: website/content/docs/nia/network-drivers/terraform-cloud.mdx
Comment (trujillo-adam) [inline website/content/docs/nia/network-drivers/terraform-cloud.mdx:72]:
```suggestion
Consul-Terraform-Sync uses Terraform input variables to reflect the latest Consul service information. They are used as parameters for your Terraform module. Input variables are dynamic and are updated by the driver throughout the runtime of Consul-Terraform-Sync.
```
After re-reading, it felt a little clunky to me. Ignore this suggestion if it wasn't bothering you.

File: website/content/docs/nia/network-drivers/terraform-cloud.mdx
Comment (trujillo-adam) [inline website/content/docs/nia/network-drivers/terraform-cloud.mdx:94]:
```suggestion
Prior to running Consul-Terraform-Sync with a Terraform Cloud driver, you will need an account and organization set up, as well as a dedicated token. We recommend using a team token that is restricted to [Manage Workspaces](https://www.terraform.io/docs/cloud/users-teams-organizations/teams.html#managing-workspace-access)-level permissions. Below are the steps for the recommended setup.
```
A bit better grammar/usage.

File: website/content/docs/nia/configuration.mdx
Comment (lornasong) [reply website/content/docs/nia/configuration.mdx:302]: Thanks for the question. The Terraform `required_providers` block doesn't have any set parameters dictated by Terraform in the sense that users can configure their providers here based on their provider's configuration parameters.

Do you have any thoughts on replacing "This is similar to the Terraform `terraform.required_providers` field to specify the source and version for each provider."  => "This can be configured the same as how you would configure the Terraform `terraform.required_providers` field to specify the source and version for each provider."

File: website/content/docs/nia/network-drivers/terraform.mdx
Comment (trujillo-adam) [reply website/content/docs/nia/network-drivers/terraform.mdx:50]: Thanks for clarifying. I was going to add a statement calling out the difference, but it might not provide enough informational value for the additional words. I guess we can update the docs if someone asks about the differences.

File: website/content/docs/nia/configuration.mdx
Comment (trujillo-adam) [reply website/content/docs/nia/configuration.mdx]: Terrific. If the note blocks get too heavy, it's probably a sign that the information should be integrated into the relevant section or even that a new subtopic should be created.

**Review by trujillo-adam — APPROVED** [review]
I added a few fixes for typos I made in my previous review. My only other suggestions is to hold off documenting and mentioning changes that will be in the next version (but that decision is probably above my pay grade) and merging the /nia/enterprise/ content into the /nia/enterprise/license page. Looks good.

File: website/content/docs/nia/enterprise/license.mdx
Comment (lornasong) [reply website/content/docs/nia/enterprise/license.mdx:33]: Thanks for catching!

File: website/content/docs/nia/enterprise/index.mdx
Comment (lornasong) [reply website/content/docs/nia/enterprise/index.mdx:5]: Gotcha! I agree that the Overview page itself doesn't provide much additional useful information. We decided to have this enterprise Overview page to since it aligns with how some of our other products' docs are set up e.g. Consul (more importantly) and Vault. I personally like having a landing page for the general enterprise section and having "License" in the side bar as a clear signpost for users looking for license information.

Do you have any additional thoughts? If we merge the two pages, would you recommend it be named "Overview" or "License"?

**Comment by lornasong** [comment]
@trujillo-adam thank you for the review and all the time you put into giving feedback on such a large set of changes in a short amount of time. We really appreciate it.

To your point about working together during drafting instead of reviewing a large PR at the end, completely agree. I'll reach out offline and loop in the team to discuss what would be a preferred process to make this happen.

We addressed some of your comments in the individual threads. Please let us know your feedback. Thank you!

FYI Update: I'm currently aiming to rebase EOD Tuesday after making final changes prior to our release

File: website/content/docs/nia/enterprise/index.mdx
Comment (trujillo-adam) [reply website/content/docs/nia/enterprise/index.mdx:5]: I see what you mean about having a landing page for CTS+ent because it would follow the docs template we employ. I still think that in this case the user experience outweighs the don't think there's enough content to merit it. In fact, I just realized that the TF Cloud section is the same as the page under Network Drivers. 

My recommendation is to completely flatten the content into "Consul-Terraform-Sync Enterprise" -- no child pages. On that page are the following topics:

* Overview
* Requirements (statement about what's different from OSS and link to /nia/installation/requirements) 
* Installation (statement about what's different from OSS and link to /nia/installation/install) 
* License
* * Setting the License
* * * Notification and Termination Behavior
* Configuration (statement about what's different from OSS and link to /nia/configure)
* * Network Drivers (link to the respective drivers pages)
* Usage (statement about what's different from OSS and link to /nia)

I think that would cover it.

Comment (lornasong) [reply website/content/docs/nia/enterprise/index.mdx:5]: Thanks for identifying additional parts of the enterprise experience that aren't called out e.g. the differences in configuring and requirements. We are actually in the process of revamping our docs overall. We'll likely revamp the enterprise content at part of the overall revamp since we've heard some other feedback as well. I'll make sure to link these recommendations so that they can be considered as part of the redesign.

Since we're aiming to release soon, I'd like to merge this PR with the current enterprise docs organization. After the release, we can iterate and improve the enterprise documentation as part of the overall documentation revamp rather than designing this twice. Please don't hesitate to let me know though if this sounds concerning to you.

To the specifics of your comment:
 - Yes, I recognize that it's not ideal and potentially confusing that the same TFC docs are in two places. This was borrowed from some existing patterns.
 - One reason we decided to have the enterprise directory was to setup for future enterprise features where each feature could have a corresponding page within the Enterprise directory.

API Tooling team, if you have differing opinions, please feel free to share.

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/439504.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit 6ee3c63c715a8ae3657ef594172f4b432d574dea onto `stable-website` succeeded!

**Pull request merged**
